lib/cfg-block.c

```diff
@@ -90,9 +90,43 @@ _validate_args_callback(gpointer k, gpointer v, gpointer user_data)
     }
 }
 
+static void
+_validate_mandatory_options_callback(gpointer parameter, gpointer default_value, gpointer user_data)
+{
+  CfgArgs *user_args = ((gpointer *) user_data)[0];
+  const gchar *reference = ((gpointer *) user_data)[1];
+  gboolean *problem_found = ((gpointer *) user_data)[2];
+
+  const gchar *value = cfg_args_get(user_args, parameter);
+
+  if (default_value == NULL && value == NULL)
+    {
+      *problem_found = TRUE;
+      msg_error("Mandatory parameter not overridden",
+                evt_tag_str("parameter", parameter),
+                evt_tag_str("reference", reference));
+    }
+}
+
+static gboolean
+_validate_mandatory_options(CfgArgs *arg_defs, CfgArgs *args, const gchar *reference)
+{
+  gboolean problem_found = FALSE;
+  gpointer validate_params[] = { args, (gchar *)reference, &problem_found };
+  cfg_args_foreach(arg_defs, _validate_mandatory_options_callback, validate_params);
+
+  if (problem_found)
+    return FALSE;
+
+  return TRUE;
+};
+
 static gboolean
 _validate_args(CfgArgs *self, CfgArgs *defs, const gchar *reference)
 {
+  if (!_validate_mandatory_options(defs, self, reference))
+    return FALSE;
+
   gboolean problem_found = FALSE;
   gpointer validate_params[] = { defs, (gchar *) reference, &problem_found };
```

Comment:
Would it be possible to hide this check also with a function just like the `_validate_args_callback` ?

Reply:
Fixed it: I named it _validate_spurious_args()